### ISSUE TITLE
Find all reference should bring me to xaml instead of the generated code

### ISSUE THREAD

**Issue by Cosifne** [dotnet/roslyn#52843]
So right now, if I have a method declared in Bar.xaml.cs file like
![image](https://user-images.githubusercontent.com/24360909/115760943-a2df7300-a356-11eb-8c64-42a2927eb5ec.png)

This method is referenced in the xaml file
![image](https://user-images.githubusercontent.com/24360909/115761000-b68ad980-a356-11eb-8605-ba98d5a145d5.png)

When press F12 in the xaml file, it would take me to the code behind, which is good.

However, if I used the FAR services, try to find all the referencing code for the method, it would take me to the generated cs code,
![image](https://user-images.githubusercontent.com/24360909/115761274-079acd80-a357-11eb-90b6-55a132948215.png)

I think the expected behavior should be, FAR should locate the method and take me to the xaml file

**Comment by CyrusNajmabadi** [comment]
FAR already calls out to a VS interface to allow other languages to take over navigation in cases like this.  It's up to Xaml to implement htat though, there's nothing for us to do here.

**Comment by Cosifne** [comment]
> FAR already calls out to a VS interface to allow other languages to take over navigation in cases like this. It's up to Xaml to implement htat though, there's nothing for us to do here.

@CyrusNajmabadi 
Is this the right place to put this issue? https://github.com/dotnet/wpf

**Comment by CyrusNajmabadi** [comment]
I'm not sure. @mgoertz-msft would likely know!

**Comment by mgoertz-msft** [comment]
@Cosifne, @CyrusNajmabadi,
There are a variety of issues at play here:

1. The .NET Core project system's IVsHierarchy does not implement IVsSymbolicNavigationNotify, and it probably never will because it's a very limited legacy interface that only ever allowed for returning a single reference. It was mostly used for XAML to provide an extra item for GoToDefinition.
2. IIRC Roslyn used to pass in the generated code file, which was checked by an optimization in our code that has been present for years. Now it passes in the .xaml.cs file so we ignore the request.
3. But even without the previous issues, method and namespace references are actually ignored during FAR in the XAML language service.

Long story short, replacing this legacy functionality with a much deeper integration between Roslyn and XAML has been on our backlog for quite some time now, but we hope to finally start this (rather big) effort in the not too distant future.

**Comment by mgexo** [comment]
P.S.

The Visual Studio IDE team could add a simple workaround for that in my oppinion, i.e. when "Got to definition"  opens the wrong file called ".g.i.cs" 

with content like this

```
      #line 398 "MainWindow.xaml"
        [System.Diagnostics.CodeAnalysis.SuppressMessageAttribute("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
        internal DentalGUIHelper.ImageAndTextToggleButton actionButtonCutView;
        
```
        
        then the IDE shall look for the "#line" tag and there is the "real" filename and then go to the line number that is written there. 
        
It is not magic and certainly possible for any next IDE update

**Comment by CyrusNajmabadi** [comment]
Dupe of https://github.com/dotnet/roslyn/issues/32643
